packages/editor-ui/src/utils/rbacUtils.ts

```diff
@@ -0,0 +1,35 @@
+import type { RouteLocationNormalized } from 'vue-router';
+import type { Resource } from '@n8n/permissions';
+
+export function inferProjectIdFromRoute(to: RouteLocationNormalized): string {
+	const routeParts = to.path.split('/');
+	const projectsIndex = routeParts.indexOf('projects');
+	const projectIdIndex = projectsIndex !== -1 ? projectsIndex + 1 : -1;
+
+	return routeParts[projectIdIndex];
+}
+
+export function inferResourceTypeFromRoute(to: RouteLocationNormalized): Resource | undefined {
+	const routeParts = to.path.split('/');
+
+	switch (true) {
```

Comment:
you could reduce some repetition with just using an `object` and a `forOf` loop
```ts
const hash = {
    workflow: 'workflows',
    sourceControl: 'source-control',
}

for (const resource of hash) {
    if(routeParts.includes(hash[resource])) {
        return resource
    }
}
```

Reply:
Awesome idea! Thanks!